Sources/NIOCertificateReloading/TimedCertificateReloader.swift

```diff
@@ -0,0 +1,421 @@
+            validatingCertificateDescription: validatingCertificateDescription,
+            validatingPrivateKeyDescription: validatingPrivateKeyDescription,
+            logger: nil
+        )
+    }
+
+    /// Initialize a new ``TimedCertificateReloader``.
+    /// - Parameters:
+    ///   - refreshInterval: The interval at which attempts to update the certificate and private key should be made.
+    ///   - certificateDescription: A ``CertificateDescription``.
+    ///   - privateKeyDescription: A ``PrivateKeyDescription``.
+    public init(
+        refreshInterval: Duration,
+        certificateDescription: CertificateDescription,
+        privateKeyDescription: PrivateKeyDescription,
+        logger: Logger? = nil
+    ) {
+        self.refreshInterval = refreshInterval
+        self.certificateDescription = certificateDescription
+        self.privateKeyDescription = privateKeyDescription
+        self.state = NIOLockedValueBox(nil)
+        self.logger = logger
+
+        // Immediately try to load the configured cert and key to avoid having to wait for the first
+        // reload loop to run.
+        // We ignore errors because this initializer tolerates not finding the certificate and/or
+        // private key on first load.
+        try? self.reloadPair()
+    }
+
+    /// Attempt to initialize a new ``TimedCertificateReloader``, but throw if the given certificate and private keys cannot be
+    /// loaded.
+    /// - Parameters:
+    ///   - refreshInterval: The interval at which attempts to update the certificate and private key should be made.
+    ///   - validatingCertificateDescription: A ``CertificateDescription``.
+    ///   - validatingPrivateKeyDescription: A ``PrivateKeyDescription``.
+    /// - Throws: If the certificate or private key cannot be loaded.
+    public init(
+        refreshInterval: Duration,
+        validatingCertificateDescription: CertificateDescription,
+        validatingPrivateKeyDescription: PrivateKeyDescription,
+        logger: Logger? = nil
+    ) throws {
+        self.refreshInterval = refreshInterval
+        self.certificateDescription = validatingCertificateDescription
+        self.privateKeyDescription = validatingPrivateKeyDescription
+        self.state = NIOLockedValueBox(nil)
+        self.logger = logger
+
+        // Immediately try to load the configured cert and key to avoid having to wait for the first
+        // reload loop to run.
+        try self.reloadPair()
+    }
+
+    /// A long-running method to run the ``TimedCertificateReloader`` and start observing updates for the certificate and
+    /// private key pair.
+    /// - Important: You *must* call this method to get certificate and key updates.
+    public func run() async throws {
+        for try await _ in AsyncTimerSequence.repeating(every: self.refreshInterval).cancelOnGracefulShutdown() {
+            do {
+                try self.reloadPair()
+            } catch {
+                self.logger?.error(
```

Comment:
```suggestion
                self.logger?.debug(
```

See https://www.swift.org/documentation/server/guides/libraries/log-levels.html